In [1]:
import pandas as pd

In [ ]:
%pip install --upgrade openpyxl

In [13]:
import pandas as pd
pd.options.display.max_columns = None
pd.options.display.max_rows = None

1. 주요 공원 현황

In [79]:
park = pd.read_csv('서울시 주요 공원현황.csv', encoding = 'cp949')
park.head(1) 

연번   공원명                                               공원개요  \
0   1  남산공원  남산공원은 도심에 위치하여 서울시민에게 맑은 공기를 제공하는 자연휴식처이며 산책 꽃...   

                                             면적        개원일  \
0  2896887㎡ 임 야 : 2454140㎡ 녹지대 및 기타시설 : 442747㎡  1968.9.10   

                                                주요시설  \
0  기반시설 : 광장 45950㎡ 도로 108530㎡ 산책로 6.7㎞ (북측:3.7㎞/...   

                                주요식물  \
0  소나무 단풍 아카시아 상수리나무 등 191종 2881870주   

                                                 안내도  \
0  http://parks.seoul.go.kr/upload/seditorMulti/2...   

                                                오시는길  \
0  명동역(남산케이블카 와룡묘 서울애니메이션센터 방면) 도보 명동역 3번출구 → 퍼시픽...   

                                             이용시참고사항  \
0  남산공원 차량통행 금지안내 2005년 5월 1일부터 남산공원에 일반승용차 택시 통행...   

                                                 이미지  지역  \
0  http://parks.seoul.go.kr/file/info/view.do?fId...  중구   

                     공원주소      관리부서          전화번호  X좌표(GRS80TM)  Y좌표(GRS80TM)  \
0  서울특별시 중구 삼일대로 231(예장동)  중부공원여가센터  02-3783-5900    198364.107    450395.554   

   X좌표(WGS84)  Y좌표(WGS84)                                             바로가기  
0  126.990377    37.55014  http://parks.seoul.go.kr/template/sub/namsan.do

In [ ]:
park.columns = park.columns.str.replace(' ', '')

In [ ]:
park = park[['공원명', '공원개요', '면적', '주요시설', '주요식물', '오시는길', '지역', 'X좌표(WGS84)', 'Y좌표(WGS84)']]
park.head()

공원명                                               공원개요  \
0    남산공원  남산공원은 도심에 위치하여 서울시민에게 맑은 공기를 제공하는 자연휴식처이며 산책 꽃...   
1  길동생태공원  길동생태공원은 생물의 서식처를 제공하고 종다양성을 증진시키며 자연생태계의 생물들을 ...   
2   서울대공원  서울대공원은 세계 각국의 야생동물들이 살아 숨 쉬는 서울동물원과 다양한 기후대의 식...   
3     서울숲  서울숲은 한강과 중랑천이 만나고 한강-용산-남찬-청계천-서울숲-한강으로 연결되는 서...   
4   월드컵공원  월드컵공원은 서울의 서쪽에 위치하여 1978년부터 1993년까지 15년간 서울시민이...   

                                             면적  \
0  2896887㎡ 임 야 : 2454140㎡ 녹지대 및 기타시설 : 442747㎡   
1                                      79258.7㎡   
2                                     9132690m²   
3                                       480994㎡   
4                                      2284085㎡   

                                                주요시설  \
0  기반시설 : 광장 45950㎡ 도로 108530㎡ 산책로 6.7㎞ (북측:3.7㎞/...   
1             생태문화센터, 탐방객안내센터, 반딧불이 체험관, 목재데크, 조류관찰대   
2  동물원 식물원 테마가든(장미원 어린이동물원 피크닉장) 치유숲 산림욕장 캠핑장 국립현...   
3  문화예술시설: 바닥분수 거울연못 군마상 야외무대 가족마당무대 생태시설: 꽃사슴방사장...   
4  평화의 공원 월드컵공원전시관(879㎡) 유니세프광장(2400㎡) 평화광장(5217㎡...   

                                                주요식물  \
0                  소나무 단풍 아카시아 상수리나무 등 191종 2881870주   
1          소나무 보리수 등 64종 31800주 산국 부들 등 138종 192800본   
2                      엘레강스야자, 용설란, 파리지옥, 네펜데스, 장미 등   
3  수 목 : 소나무 섬잣나무 계수나무 외 95종 415795주 식물원 : 선인장 등 ...   
4                                                NaN   

                                                오시는길   지역  X좌표(WGS84)  \
0  명동역(남산케이블카 와룡묘 서울애니메이션센터 방면) 도보 명동역 3번출구 → 퍼시픽...   중구  126.990377   
1  지하철: 9호선 중앙보훈병원역 3번출구에서 시내버스 환승 5호선 강동역 4번출구에서...  강동구  127.154779   
2  지하철 4호선 대공원역 하차(2번 출구) 도보 15분 정도 문의처 02)500-73...  과천시  127.019846   
3  지하철: 분당선 서울숲역 4번출구에서 도보 약 2분 2호선 뚝섬역 8번출구에서 도보...  성동구  127.041798   
4  지하철 6호선 월드컵경기장역 하차 → 1번 출구로 나온 후 직진 → 큰길(도로)이 ...  마포구  126.878907   

   Y좌표(WGS84)  
0   37.550140  
1   37.540394  
2   37.426449  
3   37.543072  
4   37.571805

In [ ]:
# park_json = park.to_json(orient='records', force_ascii=False)
# park_json

2. 공원별 세부현황

In [60]:
detail = pd.read_excel(
    "02. 2024년 공원현황 1권 제2편 공원별 세부현황.xlsx",
    sheet_name="I.시 직영", 
    engine="openpyxl",
    skiprows=2,
    usecols="A:L"
)

detail.head()

c:\Users\subin\anaconda3\envs\finfin\Lib\site-packages\openpyxl\reader\workbook.py:118: UserWarning: Print area cannot be set to Defined name: 'I.시 직영'!$A:$V.
  warn(f"Print area cannot be set to Defined name: {defn.value}.")


공원\n종류 공 원 명 위    치 Unnamed: 3  면 적 합 계\n(㎡) 토지소유별 면적\n(㎡)  Unnamed: 6  \
0    NaN   NaN    NaN        NaN           NaN           NaN         NaN   
1    NaN   NaN    NaN        NaN           NaN           NaN         NaN   
2     총계   NaN     53         12    26446424.0             국      1069.0   
3    NaN   NaN    NaN        NaN           NaN             시      3350.0   
4    NaN   NaN    NaN        NaN           NaN             구       361.0   

    Unnamed: 7  임야면적\n(GB면적)\n (㎡)  연    간\n이용자수\n(천명)  관리\n인부\n(명) 공원\n조성\n연도  
0          NaN                 NaN                 NaN          NaN        NaN  
1          NaN                 NaN                 NaN          NaN        NaN  
2  11330730.93           8104406.5            188885.0        889.0             
3  14351959.77                 NaN                 NaN          NaN        NaN  
4    150583.70           6871126.0                 NaN          NaN        NaN

In [61]:
detail.columns = detail.columns.str.replace(' ', '')
detail.columns = detail.columns.str.replace('\n', '')
detail['공원명'] = detail['공원명'].str.replace(' ', '')

detail.head()

공원종류  공원명   위치 Unnamed:3     면적합계(㎡) 토지소유별면적(㎡)  Unnamed:6    Unnamed:7  \
0  NaN  NaN  NaN       NaN         NaN        NaN        NaN          NaN   
1  NaN  NaN  NaN       NaN         NaN        NaN        NaN          NaN   
2   총계  NaN   53        12  26446424.0          국     1069.0  11330730.93   
3  NaN  NaN  NaN       NaN         NaN          시     3350.0  14351959.77   
4  NaN  NaN  NaN       NaN         NaN          구      361.0    150583.70   

   임야면적(GB면적)(㎡)  연간이용자수(천명)  관리인부(명) 공원조성연도  
0            NaN         NaN      NaN    NaN  
1            NaN         NaN      NaN    NaN  
2      8104406.5    188885.0    889.0         
3            NaN         NaN      NaN    NaN  
4      6871126.0         NaN      NaN    NaN

In [62]:
detail = detail[['공원명', '위치', '면적합계(㎡)', '연간이용자수(천명)']]
detail.head()

공원명   위치     면적합계(㎡)  연간이용자수(천명)
0  NaN  NaN         NaN         NaN
1  NaN  NaN         NaN         NaN
2  NaN   53  26446424.0    188885.0
3  NaN  NaN         NaN         NaN
4  NaN  NaN         NaN         NaN

In [63]:
detail = detail.iloc[22:]

In [ ]:
detail = detail.dropna(subset=['공원명']) ########### 수정 필요
detail.head()

공원명               위치   면적합계(㎡)  연간이용자수(천명)
22      서울숲      성동구 뚝섬로 273  237552.0      7580.0
23     (응봉)  성동구  금호 1가 1-72   70979.5      2500.0
25     시민의숲       서초구 매헌로 99  258991.0      2730.0
27       천호     강동구 올림픽로 702   26696.8      1000.0
28  서울숲(뚝섬)      성동구 뚝섬로 273  194842.0         NaN

In [65]:
len(detail)

66

In [ ]:
for col in detail.columns:
    if detail[col].dtype == 'object':
        detail[col] = detail[col].str.replace('\n', '')
detail

In [67]:
len(detail)

66

In [155]:
detail

공원명                          위치     면적합계(㎡)  연간이용자수(천명)
22                서울숲                 성동구 뚝섬로 273   237552.00      7580.0
23               (응봉)             성동구  금호 1가 1-72    70979.50      2500.0
25               시민의숲                  서초구 매헌로 99   258991.00      2730.0
27                 천호                강동구 올림픽로 702    26696.80      1000.0
28            서울숲(뚝섬)                 성동구 뚝섬로 273   194842.00         NaN
29                 율현                강남구 율현동 77-9   157535.00        72.0
30              (일자산)               강동구 천호대로 1291    79258.70        78.0
32            서울숲(기타)  습지생태원, 한강수변공원자연체험학습원, 도로 등   724104.00         NaN
41                 남산                  중구 회현1가 산1  1789845.80      1672.0
45                 낙산            종로구  동숭동 산2-1 일대     5465.80        51.0
47                 낙산                성북구 돈암 618-1     8355.00        51.0
52                  계                         NaN  1296239.70      4777.0
56                 남산   남산1근린공원(중구 남대문로5가 461 일원)   604178.00      4131.0
75                 낙산                종로구 동숭동 산2-1   150608.20       646.0
87             용산가족공원             용산구 용산6가동 68-90    75900.00       800.0
89    서울로7017(퇴계로/도로)      중구 남대문로5가 281 외    8필지     1843.20      7500.0
92   서울로7017(보행자전용도로)       중구 남대문로5가 67-4 외 37필지    10720.40         NaN
95           송현열린녹지광장                종로구 송현동 48-9    37117.00         NaN
96            율곡로상부공원           종로구 창덕궁교차로~원남동사거리     7209.00         NaN
97          광화문시민열린마당                  종로구 세종로 76     8948.00         0.0
99               서울광장                 중구 태평로1가 31    13207.00     10000.0
100             세종로공원                종로구 세종로 80-1     8868.00      1000.0
101            남산(예장)                  중구 예장동 4-1    14803.40         NaN
102    서울로7017(만리동광장)                  중구 남대문로 5가    10353.10         NaN
112            소계(조성)                         NaN  3133344.70     33087.0
116             월드컵공원          마포구 월드컵로 243-60 일대  2284085.00      6442.0
119             여의도공원               영등포구 여의공원로 68   229539.00      4745.0
120             선유도공원                영등포구 선유로 343   110407.00       792.0
122           경의선숲길공원    마포구 대흥동 일대(용산문화체육센터~가좌역)   102608.60     11498.0
125             보라매공원             동작구 여의대방로20길 33   406705.10      9610.0
128            소계(조성)                         NaN   318173.00      1199.0
130           서서울호수공원            양천구 남부순환로 64길 20   216663.00       850.0
132            문화비축기지                마포구 증산로 87일대   101510.00       349.0
141                 계                         NaN   860035.70      5150.0
145               간데메            동대문구 서울시립대로2길 59    15179.70       550.0
146             중랑캠핑숲                 중랑구 송림길 160   179666.00      1400.0
150            북서울꿈의숲                  강북구 월계로173   665190.00      3200.0
154                합계                         NaN    51146.00       830.0
155             서울창포원                 도봉구 마들로 916    51146.00       830.0
160             경춘선숲길                  노원구 하계동 일대   178661.00      3000.0
168             계(조성)                         NaN   706306.90      3971.0
172             서울식물원                 강서구 마곡동 812   505350.90      3611.0
173             푸른수목원              구로구 연동로 240 일대   200956.00       360.0
183             서울대공원                과천시 막계 159-1  6670000.00      1479.0
194            어린이대공원                 광진구 능동로 216   560552.00      6845.0
198          추모공원(조성)                  서초구 원지동 76   127836.00         NaN
201           청계천(합계)                   종로1가동~사근동   375705.00     25000.0
202          청계천(종로구)                  종로1가동~숭인2동    40256.70         NaN
203           청계천(중구)                  태평로1가동~황학동    27801.09         NaN
204          청계천(동대문)                    신설동~용두1동    20596.00         NaN
205          청계천(성동구)                   왕십리1동~사근동   287051.21         NaN
208            한강공원양화                        영등포구   361628.00      2458.0
209            한강공원망원                         마포구   422347.00      2521.0
210           한강공원여의도 

3. 공원 시설물 현황

In [138]:
facilities = pd.read_excel(
    "04. 2024년 공원현황 2권 공원시설물현황.xlsx",
    sheet_name="시직영(사업소)", 
    header=[0, 1, 2],
    engine="openpyxl",
    skiprows=2,
)

facilities.head()

공   원   명 기   반   시   설                                       \
  Unnamed: 0_level_1         광   장             도   로              산책등산로   
  Unnamed: 0_level_2            개소          ㎡     노선           ㎡     노선   
0           시 직영(총계)         176.0  409216.83  190.0  1059990.15  511.0   
1       동부공원여가센터(총계)           8.0   27946.00   15.0    99797.20    6.0   
2      도시계획시설(공원) 합계           8.0   27946.00   15.0    99797.20    6.0   
3        생활권공원(근린) 계           8.0   27946.00   15.0    99797.20    6.0   
4              시민의 숲           1.0    3179.00    3.0    20060.00    2.0   

            조   경   시   설                                               \
                   관상용식수대                그늘\n시렁\n(개소) 연   못              
          ㎞            개소        ㎡ Unnamed: 9_level_2    개소          ㎡   
0  212.8399         734.0  15367.0              376.0  56.0  431138.49   
1   13.1000           0.0      0.0               74.0   8.0   35137.00   
2   13.1000           0.0      0.0               74.0   8.0   35137.00   
3   13.1000           0.0      0.0               74.0   8.0   35137.00   
4    4.2000           NaN      NaN                3.0   1.0    2604.00   

                                                       휴   양   시   설  \
              분수\n(기)             조각\n(소) 기   타            야영장\n야유회장   
  Unnamed: 12_level_2 Unnamed: 13_level_2     종      기            개소   
0                70.0                  58  45.0  156.0             7   
1                15.0                  11   3.0    4.0             0   
2                15.0                  11   3.0    4.0             0   
3                15.0                  11   3.0    4.0             0   
4                 1.0                  10   3.0    4.0                 

                                                                   \
             정   자(㎡)          노인정(㎡)                 야외\n탁자\n(기)   
           ㎡        동     건축면적      동 건축면적(㎡) Unnamed: 22_level_2   
0  135360.98   1678.0  2752.32    2.0  432.03              2215.0   
1       0.00     13.0   317.25    1.0  218.03                77.0   
2       0.00     13.0   317.25    1.0  218.03                77.0   
3       0.00     13.0   317.25    1.0  218.03                77.0   
4        NaN     10.0   216.00    NaN     NaN                 9.0   

                                         유   희   시   설                      \
             장의자\n(기) 기   타                    그네\n(기)         미끄\n럼틀\n(기)   
  Unnamed: 23_level_2     종      기 Unnamed: 26_level_2 Unnamed: 27_level_2   
0              8618.0  40.0  657.0                38.0                13.0   
1               904.0   3.0   35.0                 4.0                 5.0   
2               904.0   3.0   35.0                 4.0                 5.0   
3               904.0   3.0   35.0                 4.0                 5.0   
4               280.0   3.0   35.0                 NaN                 NaN   

                                                               \
           모래\n밭\n(기)             시소\n(기)          정글\n짐\n(기)   
  Unnamed: 28_level_2 Unnamed: 29_level_2 Unnamed: 30_level_2   
0                35.0                20.0                 5.0   
1                 6.0                 0.0                 0.0   
2                 6.0                 0.0                 0.0   
3                 6.0                 0.0                 0.0   
4                 1.0                 NaN                 NaN   

                                                                     \
          사닥\n다리\n(기)         조합\n놀이\n(기)  유료시설        기   타          
  Unnamed: 31_level_2 Unnamed: 32_level_2     종     기      종      기   
0                 0.0                55.0  15.0  18.0  111.0  201.0   
1                 0.0                11.0   0.0   0.0   27.0   37.0   
2                 0.0                11.0   0.0   0.0   27.0   37.0   
3                 0.0                11.0   0.0   0.0   27.0   37.0   
4                 NaN                 1.0   

In [139]:
facilities = facilities.iloc[4:]
facilities.head()

공   원   명 기   반   시   설                                    \
  Unnamed: 0_level_1         광   장         도   로          산책등산로        
  Unnamed: 0_level_2            개소       ㎡    노선        ㎡    노선    ㎞   
4              시민의 숲           1.0  3179.0   3.0  20060.0   2.0  4.2   
5              천   호           1.0  1180.0   NaN      NaN   NaN  NaN   
6              응   봉           1.0  1692.0   1.0   1380.0   1.0  6.6   
7        서울숲(생활권-근린)           1.0  8462.0   6.0  38120.0   1.0  1.2   
8         뚝섬(생활권-근린)           1.0  7032.0   3.0  17484.2   1.0  0.7   

  조   경   시   설                                                            \
         관상용식수대           그늘\n시렁\n(개소) 연   못                      분수\n(기)   
             개소   ㎡ Unnamed: 9_level_2    개소        ㎡ Unnamed: 12_level_2   
4           NaN NaN                3.0   1.0   2604.0                 1.0   
5           NaN NaN                NaN   1.0    674.0                 1.0   
6           NaN NaN               20.0   NaN      NaN                 NaN   
7           NaN NaN               23.0   2.0  19099.0                11.0   
8           NaN NaN                NaN   2.0   8286.0                 NaN   

                                 휴   양   시   설                             \
              조각\n(소) 기   타          야영장\n야유회장     정   자(㎡)        노인정(㎡)   
  Unnamed: 13_level_2     종    기            개소   ㎡        동   건축면적      동   
4                  10   3.0  4.0               NaN     10.0  216.0    NaN   
5                 NaN   NaN  NaN           NaN NaN      1.0   36.0    NaN   
6                 NaN   NaN  NaN           NaN NaN      NaN    NaN    1.0   
7                   1   NaN  NaN           NaN NaN      NaN    NaN    NaN   
8                 NaN   NaN  NaN           NaN NaN      NaN    NaN    NaN   

                                                               \
                  야외\n탁자\n(기)            장의자\n(기) 기   타         
  건축면적(㎡) Unnamed: 22_level_2 Unnamed: 23_level_2     종     기   
4     NaN                 9.0               280.0   3.0  35.0   
5     NaN                18.0                81.0   NaN   NaN   
6  218.03                15.0                98.0   NaN   NaN   
7     NaN                25.0               320.0   NaN   NaN   
8     NaN                10.0                30.0   NaN   NaN   

        유   희   시   설                                          \
              그네\n(기)         미끄\n럼틀\n(기)          모래\n밭\n(기)   
  Unnamed: 26_level_2 Unnamed: 27_level_2 Unnamed: 28_level_2   
4                 NaN                 NaN                 1.0   
5                 NaN                 NaN                 NaN   
6                 2.0                 2.0                 1.0   
7                 NaN                 3.0                 3.0   
8                 NaN                 NaN                 NaN   

                                                               \
              시소\n(기)          정글\n짐\n(기)         사닥\n다리\n(기)   
  Unnamed: 29_level_2 Unnamed: 30_level_2 Unnamed: 31_level_2   
4                 NaN                 NaN                 NaN   
5                 NaN                 NaN                 NaN   
6                 NaN                 NaN                 NaN   
7                 NaN                 NaN                 NaN   
8                 NaN                 NaN                 NaN   

                                                     공   원   명 운   동   시   설  \
          조합\n놀이\n(기) 유료시설     기   타       Unnamed: 37_level_1           운동장   
  Unnamed: 32_level_2    종   기     종     기 Unnamed: 37_level_2            개소   
4                 1.0  NaN NaN   2.0   2.0               시민의 숲           1.0   
5                 NaN  NaN NaN   NaN   NaN               천   호           NaN   
6                 1.0  NaN NaN   3.0   6.0               응   봉           NaN   
7                 2.0  NaN NaN  19.0  20.0         서울숲(생활권-근린)           1.0   
8                 NaN  NaN NaN   NaN   NaN          뚝섬(생활권-근린)           NaN 

In [140]:
# 공원명 중복 열 삭제
columns_to_drop_by_index = [37, 73, 102]

facilities = facilities.drop(facilities.columns[columns_to_drop_by_index], axis=1)
facilities.head()

공   원   명 기   반   시   설                                    \
  Unnamed: 0_level_1         광   장         도   로          산책등산로        
  Unnamed: 0_level_2            개소       ㎡    노선        ㎡    노선    ㎞   
4              시민의 숲           1.0  3179.0   3.0  20060.0   2.0  4.2   
5              천   호           1.0  1180.0   NaN      NaN   NaN  NaN   
6              응   봉           1.0  1692.0   1.0   1380.0   1.0  6.6   
7        서울숲(생활권-근린)           1.0  8462.0   6.0  38120.0   1.0  1.2   
8         뚝섬(생활권-근린)           1.0  7032.0   3.0  17484.2   1.0  0.7   

  조   경   시   설                                                            \
         관상용식수대           그늘\n시렁\n(개소) 연   못                      분수\n(기)   
             개소   ㎡ Unnamed: 9_level_2    개소        ㎡ Unnamed: 12_level_2   
4           NaN NaN                3.0   1.0   2604.0                 1.0   
5           NaN NaN                NaN   1.0    674.0                 1.0   
6           NaN NaN               20.0   NaN      NaN                 NaN   
7           NaN NaN               23.0   2.0  19099.0                11.0   
8           NaN NaN                NaN   2.0   8286.0                 NaN   

                                 휴   양   시   설                             \
              조각\n(소) 기   타          야영장\n야유회장     정   자(㎡)        노인정(㎡)   
  Unnamed: 13_level_2     종    기            개소   ㎡        동   건축면적      동   
4                  10   3.0  4.0               NaN     10.0  216.0    NaN   
5                 NaN   NaN  NaN           NaN NaN      1.0   36.0    NaN   
6                 NaN   NaN  NaN           NaN NaN      NaN    NaN    1.0   
7                   1   NaN  NaN           NaN NaN      NaN    NaN    NaN   
8                 NaN   NaN  NaN           NaN NaN      NaN    NaN    NaN   

                                                               \
                  야외\n탁자\n(기)            장의자\n(기) 기   타         
  건축면적(㎡) Unnamed: 22_level_2 Unnamed: 23_level_2     종     기   
4     NaN                 9.0               280.0   3.0  35.0   
5     NaN                18.0                81.0   NaN   NaN   
6  218.03                15.0                98.0   NaN   NaN   
7     NaN                25.0               320.0   NaN   NaN   
8     NaN                10.0                30.0   NaN   NaN   

        유   희   시   설                                          \
              그네\n(기)         미끄\n럼틀\n(기)          모래\n밭\n(기)   
  Unnamed: 26_level_2 Unnamed: 27_level_2 Unnamed: 28_level_2   
4                 NaN                 NaN                 1.0   
5                 NaN                 NaN                 NaN   
6                 2.0                 2.0                 1.0   
7                 NaN                 3.0                 3.0   
8                 NaN                 NaN                 NaN   

                                                               \
              시소\n(기)          정글\n짐\n(기)         사닥\n다리\n(기)   
  Unnamed: 29_level_2 Unnamed: 30_level_2 Unnamed: 31_level_2   
4                 NaN                 NaN                 NaN   
5                 NaN                 NaN                 NaN   
6                 NaN                 NaN                 NaN   
7                 NaN                 NaN                 NaN   
8                 NaN                 NaN                 NaN   

                                           운   동   시   설                     \
          조합\n놀이\n(기) 유료시설     기   타                 운동장         테니스장         
  Unnamed: 32_level_2    종   기     종     기            개소       ㎡   개소    면수   
4                 1.0  NaN NaN   2.0   2.0           1.0  3050.0  1.0  11.0   
5                 NaN  NaN NaN   NaN   NaN           NaN     NaN  NaN   NaN   
6                 1.0  NaN NaN   3.0   6.0           NaN     NaN  1.0   2.0   
7                 2.0  NaN NaN  19.0  20.0           1.0  1558.0  1.0   5.0   
8                 NaN  NaN NaN   NaN   NaN           NaN     NaN  NaN   NaN   

    

In [141]:
# 관리시설 ~ 삭제
facilities = facilities.drop(columns=['관   리   시   설', '적  법  건  물', '부  적  격  건  물'], level=0)
facilities.head()

공   원   명 기   반   시   설                                    \
  Unnamed: 0_level_1         광   장         도   로          산책등산로        
  Unnamed: 0_level_2            개소       ㎡    노선        ㎡    노선    ㎞   
4              시민의 숲           1.0  3179.0   3.0  20060.0   2.0  4.2   
5              천   호           1.0  1180.0   NaN      NaN   NaN  NaN   
6              응   봉           1.0  1692.0   1.0   1380.0   1.0  6.6   
7        서울숲(생활권-근린)           1.0  8462.0   6.0  38120.0   1.0  1.2   
8         뚝섬(생활권-근린)           1.0  7032.0   3.0  17484.2   1.0  0.7   

  조   경   시   설                                                            \
         관상용식수대           그늘\n시렁\n(개소) 연   못                      분수\n(기)   
             개소   ㎡ Unnamed: 9_level_2    개소        ㎡ Unnamed: 12_level_2   
4           NaN NaN                3.0   1.0   2604.0                 1.0   
5           NaN NaN                NaN   1.0    674.0                 1.0   
6           NaN NaN               20.0   NaN      NaN                 NaN   
7           NaN NaN               23.0   2.0  19099.0                11.0   
8           NaN NaN                NaN   2.0   8286.0                 NaN   

                                 휴   양   시   설                             \
              조각\n(소) 기   타          야영장\n야유회장     정   자(㎡)        노인정(㎡)   
  Unnamed: 13_level_2     종    기            개소   ㎡        동   건축면적      동   
4                  10   3.0  4.0               NaN     10.0  216.0    NaN   
5                 NaN   NaN  NaN           NaN NaN      1.0   36.0    NaN   
6                 NaN   NaN  NaN           NaN NaN      NaN    NaN    1.0   
7                   1   NaN  NaN           NaN NaN      NaN    NaN    NaN   
8                 NaN   NaN  NaN           NaN NaN      NaN    NaN    NaN   

                                                               \
                  야외\n탁자\n(기)            장의자\n(기) 기   타         
  건축면적(㎡) Unnamed: 22_level_2 Unnamed: 23_level_2     종     기   
4     NaN                 9.0               280.0   3.0  35.0   
5     NaN                18.0                81.0   NaN   NaN   
6  218.03                15.0                98.0   NaN   NaN   
7     NaN                25.0               320.0   NaN   NaN   
8     NaN                10.0                30.0   NaN   NaN   

        유   희   시   설                                          \
              그네\n(기)         미끄\n럼틀\n(기)          모래\n밭\n(기)   
  Unnamed: 26_level_2 Unnamed: 27_level_2 Unnamed: 28_level_2   
4                 NaN                 NaN                 1.0   
5                 NaN                 NaN                 NaN   
6                 2.0                 2.0                 1.0   
7                 NaN                 3.0                 3.0   
8                 NaN                 NaN                 NaN   

                                                               \
              시소\n(기)          정글\n짐\n(기)         사닥\n다리\n(기)   
  Unnamed: 29_level_2 Unnamed: 30_level_2 Unnamed: 31_level_2   
4                 NaN                 NaN                 NaN   
5                 NaN                 NaN                 NaN   
6                 NaN                 NaN                 NaN   
7                 NaN                 NaN                 NaN   
8                 NaN                 NaN                 NaN   

                                           운   동   시   설                     \
          조합\n놀이\n(기) 유료시설     기   타                 운동장         테니스장         
  Unnamed: 32_level_2    종   기     종     기            개소       ㎡   개소    면수   
4                 1.0  NaN NaN   2.0   2.0           1.0  3050.0  1.0  11.0   
5                 NaN  NaN NaN   NaN   NaN           NaN     NaN  NaN   NaN   
6                 1.0  NaN NaN   3.0   6.0           NaN     NaN  1.0   2.0   
7                 2.0  NaN NaN  19.0  20.0           1.0  1558.0  1.0   5.0   
8                 NaN  NaN NaN   NaN   NaN           NaN     NaN  NaN   NaN   

    

In [142]:
# 시설 인덱스 삭제
facilities.columns = facilities.columns.droplevel(0)
facilities.head()

Unnamed: 0_level_1 광   장         도   로          산책등산로      관상용식수대      \
  Unnamed: 0_level_2    개소       ㎡    노선        ㎡    노선    ㎞     개소   ㎡   
4              시민의 숲   1.0  3179.0   3.0  20060.0   2.0  4.2    NaN NaN   
5              천   호   1.0  1180.0   NaN      NaN   NaN  NaN    NaN NaN   
6              응   봉   1.0  1692.0   1.0   1380.0   1.0  6.6    NaN NaN   
7        서울숲(생활권-근린)   1.0  8462.0   6.0  38120.0   1.0  1.2    NaN NaN   
8         뚝섬(생활권-근린)   1.0  7032.0   3.0  17484.2   1.0  0.7    NaN NaN   

        그늘\n시렁\n(개소) 연   못                      분수\n(기)             조각\n(소)  \
  Unnamed: 9_level_2    개소        ㎡ Unnamed: 12_level_2 Unnamed: 13_level_2   
4                3.0   1.0   2604.0                 1.0                  10   
5                NaN   1.0    674.0                 1.0                 NaN   
6               20.0   NaN      NaN                 NaN                 NaN   
7               23.0   2.0  19099.0                11.0                   1   
8                NaN   2.0   8286.0                 NaN                 NaN   

  기   타      야영장\n야유회장     정   자(㎡)        노인정(㎡)                 야외\n탁자\n(기)  \
      종    기        개소   ㎡        동   건축면적      동 건축면적(㎡) Unnamed: 22_level_2   
4   3.0  4.0           NaN     10.0  216.0    NaN     NaN                 9.0   
5   NaN  NaN       NaN NaN      1.0   36.0    NaN     NaN                18.0   
6   NaN  NaN       NaN NaN      NaN    NaN    1.0  218.03                15.0   
7   NaN  NaN       NaN NaN      NaN    NaN    NaN     NaN                25.0   
8   NaN  NaN       NaN NaN      NaN    NaN    NaN     NaN                10.0   

             장의자\n(기) 기   타                   그네\n(기)         미끄\n럼틀\n(기)  \
  Unnamed: 23_level_2     종     기 Unnamed: 26_level_2 Unnamed: 27_level_2   
4               280.0   3.0  35.0                 NaN                 NaN   
5                81.0   NaN   NaN                 NaN                 NaN   
6                98.0   NaN   NaN                 2.0                 2.0   
7               320.0   NaN   NaN                 NaN                 3.0   
8                30.0   NaN   NaN                 NaN                 NaN   

           모래\n밭\n(기)             시소\n(기)          정글\n짐\n(기)  \
  Unnamed: 28_level_2 Unnamed: 29_level_2 Unnamed: 30_level_2   
4                 1.0                 NaN                 NaN   
5                 NaN                 NaN                 NaN   
6                 1.0                 NaN                 NaN   
7                 3.0                 NaN                 NaN   
8                 NaN                 NaN                 NaN   

          사닥\n다리\n(기)         조합\n놀이\n(기) 유료시설     기   타        운동장          \
  Unnamed: 31_level_2 Unnamed: 32_level_2    종   기     종     기   개소       ㎡   
4                 NaN                 1.0  NaN NaN   2.0   2.0  1.0  3050.0   
5                 NaN                 NaN  NaN NaN   NaN   NaN  NaN     NaN   
6                 NaN                 1.0  NaN NaN   3.0   6.0  NaN     NaN   
7                 NaN                 2.0  NaN NaN  19.0  20.0  1.0  1558.0   
8                 NaN                 NaN  NaN NaN   NaN   NaN  NaN     NaN   

  테니스장       수영장(실내,㎡)         수영장(실외)              궁도\n장\n(소)  \
    개소    면수        개소 건축면적(㎡)      개소   ㎡ Unnamed: 46_level_2   
4  1.0  11.0       NaN     NaN     NaN NaN                 NaN   
5  NaN   NaN       NaN     NaN     NaN NaN                 NaN   
6  1.0   2.0       NaN     NaN     NaN NaN                 NaN   
7  1.0   5.0       NaN     NaN     NaN NaN                 NaN   
8  NaN   NaN       NaN     NaN     NaN NaN                 NaN   

           야구\n장\n(소)          축구\n장\n(소)          농구\n장\n(소)  \
  Unnamed: 47_level_2 Unnamed: 48_level_2 Unnamed: 49_level_2   
4                 NaN                 NaN                 2.0   
5                 NaN                 NaN                 2.0   
6                 NaN                 1.0                 NaN   
7               

In [143]:
# 기타 행 삭제
mask = facilities.columns.get_level_values(0) != '기   타'

facilities = facilities.loc[:, mask]
facilities.head()

Unnamed: 0_level_1 광   장         도   로          산책등산로      관상용식수대      \
  Unnamed: 0_level_2    개소       ㎡    노선        ㎡    노선    ㎞     개소   ㎡   
4              시민의 숲   1.0  3179.0   3.0  20060.0   2.0  4.2    NaN NaN   
5              천   호   1.0  1180.0   NaN      NaN   NaN  NaN    NaN NaN   
6              응   봉   1.0  1692.0   1.0   1380.0   1.0  6.6    NaN NaN   
7        서울숲(생활권-근린)   1.0  8462.0   6.0  38120.0   1.0  1.2    NaN NaN   
8         뚝섬(생활권-근린)   1.0  7032.0   3.0  17484.2   1.0  0.7    NaN NaN   

        그늘\n시렁\n(개소) 연   못                      분수\n(기)             조각\n(소)  \
  Unnamed: 9_level_2    개소        ㎡ Unnamed: 12_level_2 Unnamed: 13_level_2   
4                3.0   1.0   2604.0                 1.0                  10   
5                NaN   1.0    674.0                 1.0                 NaN   
6               20.0   NaN      NaN                 NaN                 NaN   
7               23.0   2.0  19099.0                11.0                   1   
8                NaN   2.0   8286.0                 NaN                 NaN   

  야영장\n야유회장     정   자(㎡)        노인정(㎡)                 야외\n탁자\n(기)  \
         개소   ㎡        동   건축면적      동 건축면적(㎡) Unnamed: 22_level_2   
4           NaN     10.0  216.0    NaN     NaN                 9.0   
5       NaN NaN      1.0   36.0    NaN     NaN                18.0   
6       NaN NaN      NaN    NaN    1.0  218.03                15.0   
7       NaN NaN      NaN    NaN    NaN     NaN                25.0   
8       NaN NaN      NaN    NaN    NaN     NaN                10.0   

             장의자\n(기)             그네\n(기)         미끄\n럼틀\n(기)  \
  Unnamed: 23_level_2 Unnamed: 26_level_2 Unnamed: 27_level_2   
4               280.0                 NaN                 NaN   
5                81.0                 NaN                 NaN   
6                98.0                 2.0                 2.0   
7               320.0                 NaN                 3.0   
8                30.0                 NaN                 NaN   

           모래\n밭\n(기)             시소\n(기)          정글\n짐\n(기)  \
  Unnamed: 28_level_2 Unnamed: 29_level_2 Unnamed: 30_level_2   
4                 1.0                 NaN                 NaN   
5                 NaN                 NaN                 NaN   
6                 1.0                 NaN                 NaN   
7                 3.0                 NaN                 NaN   
8                 NaN                 NaN                 NaN   

          사닥\n다리\n(기)         조합\n놀이\n(기) 유료시설      운동장         테니스장        \
  Unnamed: 31_level_2 Unnamed: 32_level_2    종   기   개소       ㎡   개소    면수   
4                 NaN                 1.0  NaN NaN  1.0  3050.0  1.0  11.0   
5                 NaN                 NaN  NaN NaN  NaN     NaN  NaN   NaN   
6                 NaN                 1.0  NaN NaN  NaN     NaN  1.0   2.0   
7                 NaN                 2.0  NaN NaN  1.0  1558.0  1.0   5.0   
8                 NaN                 NaN  NaN NaN  NaN     NaN  NaN   NaN   

  수영장(실내,㎡)         수영장(실외)              궁도\n장\n(소)          야구\n장\n(소)  \
         개소 건축면적(㎡)      개소   ㎡ Unnamed: 46_level_2 Unnamed: 47_level_2   
4       NaN     NaN     NaN NaN                 NaN                 NaN   
5       NaN     NaN     NaN NaN                 NaN                 NaN   
6       NaN     NaN     NaN NaN                 NaN                 NaN   
7       NaN     NaN     NaN NaN                 NaN                 NaN   
8       NaN     NaN     NaN NaN                 NaN                 NaN   

           축구\n장\n(소)          농구\n장\n(소)          배구\n장\n(소) 배드민턴장         \
  Unnamed: 48_level_2 Unnamed: 49_level_2 Unnamed: 50_level_2    개소  면적(㎡)   
4                 NaN                 2.0                 5.0   NaN    NaN   
5                 NaN                 2.0                 NaN   2.0  600.0   
6                 1.0                 NaN                 NaN   1.0    6.0   
7                 NaN                 1.0               

In [146]:
# 	Unnamed: 0_level_2 정리
columns_to_remove = ['㎡', '건축면적(㎡)', '면적(㎡)', '㎞', '건축면적', '기', '타석', '주차대수', '면수']

mask = ~facilities.columns.get_level_values(1).isin(columns_to_remove)

facilities = facilities.loc[:, mask]

facilities.columns = facilities.columns.droplevel(1)

facilities.head()

Unnamed: 0_level_1  광   장  도   로  산책등산로  관상용식수대  그늘\n시렁\n(개소)  연   못  \
4              시민의 숲    1.0    3.0    2.0     NaN           3.0    1.0   
5              천   호    1.0    NaN    NaN     NaN           NaN    1.0   
6              응   봉    1.0    1.0    1.0     NaN          20.0    NaN   
7        서울숲(생활권-근린)    1.0    6.0    1.0     NaN          23.0    2.0   
8         뚝섬(생활권-근린)    1.0    3.0    1.0     NaN           NaN    2.0   

   분수\n(기) 조각\n(소) 야영장\n야유회장  정   자(㎡)  노인정(㎡)  야외\n탁자\n(기)  장의자\n(기)  \
4      1.0      10                10.0     NaN          9.0     280.0   
5      1.0     NaN       NaN       1.0     NaN         18.0      81.0   
6      NaN     NaN       NaN       NaN     1.0         15.0      98.0   
7     11.0       1       NaN       NaN     NaN         25.0     320.0   
8      NaN     NaN       NaN       NaN     NaN         10.0      30.0   

   그네\n(기)  미끄\n럼틀\n(기)  모래\n밭\n(기)  시소\n(기)  정글\n짐\n(기)  사닥\n다리\n(기)  \
4      NaN          NaN         1.0      NaN         NaN          NaN   
5      NaN          NaN         NaN      NaN         NaN          NaN   
6      2.0          2.0         1.0      NaN         NaN          NaN   
7      NaN          3.0         3.0      NaN         NaN          NaN   
8      NaN          NaN         NaN      NaN         NaN          NaN   

   조합\n놀이\n(기)  유료시설  운동장  테니스장  수영장(실내,㎡)  수영장(실외)  궁도\n장\n(소)  야구\n장\n(소)  \
4          1.0   NaN  1.0   1.0        NaN      NaN         NaN         NaN   
5          NaN   NaN  NaN   NaN        NaN      NaN         NaN         NaN   
6          1.0   NaN  NaN   1.0        NaN      NaN         NaN         NaN   
7          2.0   NaN  1.0   1.0        NaN      NaN         NaN         NaN   
8          NaN   NaN  NaN   NaN        NaN      NaN         NaN         NaN   

   축구\n장\n(소)  농구\n장\n(소)  배구\n장\n(소)  배드민턴장 골프연습장  게이\n트볼\n(소) 체력단련시설  동물원  \
4         NaN         2.0         5.0    NaN   NaN          NaN      3  NaN   
5         NaN         2.0         NaN    2.0   NaN          NaN      6  NaN   
6         1.0         NaN         NaN    1.0   NaN          2.0      9  NaN   
7         NaN         1.0         NaN    2.0   NaN          1.0      2  NaN   
8         NaN         NaN         NaN    NaN   NaN          NaN      -  NaN   

   식물원  도서관\n독서실  문화회관  야외\n음악당\n(소) 동상\n(기)  탑기\n념비\n(기)  주   차   장  \
4  NaN       NaN   NaN           NaN       1          5.0        1.0   
5  NaN       1.0   NaN           1.0     NaN          NaN        NaN   
6  NaN       NaN   NaN           NaN     NaN          NaN        NaN   
7  NaN       NaN   NaN           1.0                  NaN        2.0   
8  1.0       NaN   NaN           NaN     NaN          NaN        NaN   

   휴게음식점(식당)  매   점  자판기  화장실(수세식) 발효식\n화장실\n(기)  간   이\n화장실\n(기) 공중\n전화\n(기)  \
4        NaN    1.0  NaN       5.0           NaN              NaN           2   
5        NaN    NaN  NaN       1.0           NaN              NaN           -   
6        NaN    NaN  NaN       2.0           NaN              NaN         NaN   
7        1.0    2.0  NaN       7.0           NaN              NaN           4   
8        NaN    1.0  NaN       3.0           NaN              NaN           1   

   음수대\n(기)  샘터\n(소)  시계탑\n(기)  
4       7.0      NaN       NaN  
5       2.0      NaN       NaN  
6       2.0      NaN       NaN  
7       7.0      NaN       1.0  
8       1.0      NaN       NaN

In [149]:
# Unnamed: 0_level_1 이름 변경
facilities = facilities.rename(columns={
    'Unnamed: 0_level_1' : '공원명',
    })
facilities.head()

공원명  광   장  도   로  산책등산로  관상용식수대  그늘\n시렁\n(개소)  연   못  분수\n(기)  \
4          시민의 숲    1.0    3.0    2.0     NaN           3.0    1.0      1.0   
5          천   호    1.0    NaN    NaN     NaN           NaN    1.0      1.0   
6          응   봉    1.0    1.0    1.0     NaN          20.0    NaN      NaN   
7    서울숲(생활권-근린)    1.0    6.0    1.0     NaN          23.0    2.0     11.0   
8     뚝섬(생활권-근린)    1.0    3.0    1.0     NaN           NaN    2.0      NaN   

  조각\n(소) 야영장\n야유회장  정   자(㎡)  노인정(㎡)  야외\n탁자\n(기)  장의자\n(기)  그네\n(기)  \
4      10                10.0     NaN          9.0     280.0      NaN   
5     NaN       NaN       1.0     NaN         18.0      81.0      NaN   
6     NaN       NaN       NaN     1.0         15.0      98.0      2.0   
7       1       NaN       NaN     NaN         25.0     320.0      NaN   
8     NaN       NaN       NaN     NaN         10.0      30.0      NaN   

   미끄\n럼틀\n(기)  모래\n밭\n(기)  시소\n(기)  정글\n짐\n(기)  사닥\n다리\n(기)  조합\n놀이\n(기)  \
4          NaN         1.0      NaN         NaN          NaN          1.0   
5          NaN         NaN      NaN         NaN          NaN          NaN   
6          2.0         1.0      NaN         NaN          NaN          1.0   
7          3.0         3.0      NaN         NaN          NaN          2.0   
8          NaN         NaN      NaN         NaN          NaN          NaN   

   유료시설  운동장  테니스장  수영장(실내,㎡)  수영장(실외)  궁도\n장\n(소)  야구\n장\n(소)  축구\n장\n(소)  \
4   NaN  1.0   1.0        NaN      NaN         NaN         NaN         NaN   
5   NaN  NaN   NaN        NaN      NaN         NaN         NaN         NaN   
6   NaN  NaN   1.0        NaN      NaN         NaN         NaN         1.0   
7   NaN  1.0   1.0        NaN      NaN         NaN         NaN         NaN   
8   NaN  NaN   NaN        NaN      NaN         NaN         NaN         NaN   

   농구\n장\n(소)  배구\n장\n(소)  배드민턴장 골프연습장  게이\n트볼\n(소) 체력단련시설  동물원  식물원  \
4         2.0         5.0    NaN   NaN          NaN      3  NaN  NaN   
5         2.0         NaN    2.0   NaN          NaN      6  NaN  NaN   
6         NaN         NaN    1.0   NaN          2.0      9  NaN  NaN   
7         1.0         NaN    2.0   NaN          1.0      2  NaN  NaN   
8         NaN         NaN    NaN   NaN          NaN      -  NaN  1.0   

   도서관\n독서실  문화회관  야외\n음악당\n(소) 동상\n(기)  탑기\n념비\n(기)  주   차   장  휴게음식점(식당)  \
4       NaN   NaN           NaN       1          5.0        1.0        NaN   
5       1.0   NaN           1.0     NaN          NaN        NaN        NaN   
6       NaN   NaN           NaN     NaN          NaN        NaN        NaN   
7       NaN   NaN           1.0                  NaN        2.0        1.0   
8       NaN   NaN           NaN     NaN          NaN        NaN        NaN   

   매   점  자판기  화장실(수세식) 발효식\n화장실\n(기)  간   이\n화장실\n(기) 공중\n전화\n(기)  음수대\n(기)  \
4    1.0  NaN       5.0           NaN              NaN           2       7.0   
5    NaN  NaN       1.0           NaN              NaN           -       2.0   
6    NaN  NaN       2.0           NaN              NaN         NaN       2.0   
7    2.0  NaN       7.0           NaN              NaN           4       7.0   
8    1.0  NaN       3.0           NaN              NaN           1       1.0   

   샘터\n(소)  시계탑\n(기)  
4      NaN       NaN  
5      NaN       NaN  
6      NaN       NaN  
7      NaN       1.0  
8      NaN       NaN

In [150]:
facilities.columns = facilities.columns.str.replace('\n', '')
facilities.columns = facilities.columns.str.replace(' ', '')
facilities['공원명'] = facilities['공원명'].str.replace(' ', '')

In [ ]:
import re

facilities.columns = facilities.columns.str.replace(r'\(.*\)', '', regex=True)
facilities.head()

In [ ]:
facilities = facilities.fillna(0)


In [154]:
facilities.head()

공원명   광장   도로  산책등산로  관상용식수대  그늘시렁   연못    분수  조각 야영장야유회장    정자  \
4         시민의숲  1.0  3.0    2.0     0.0   3.0  1.0   1.0  10          10.0   
5           천호  1.0  0.0    0.0     0.0   0.0  1.0   1.0   0       0   1.0   
6           응봉  1.0  1.0    1.0     0.0  20.0  0.0   0.0   0       0   0.0   
7  서울숲(생활권-근린)  1.0  6.0    1.0     0.0  23.0  2.0  11.0   1       0   0.0   
8   뚝섬(생활권-근린)  1.0  3.0    1.0     0.0   0.0  2.0   0.0   0       0   0.0   

   노인정  야외탁자    장의자   그네  미끄럼틀  모래밭   시소  정글짐  사닥다리  조합놀이  유료시설  운동장  테니스장  \
4  0.0   9.0  280.0  0.0   0.0  1.0  0.0  0.0   0.0   1.0   0.0  1.0   1.0   
5  0.0  18.0   81.0  0.0   0.0  0.0  0.0  0.0   0.0   0.0   0.0  0.0   0.0   
6  1.0  15.0   98.0  2.0   2.0  1.0  0.0  0.0   0.0   1.0   0.0  0.0   1.0   
7  0.0  25.0  320.0  0.0   3.0  3.0  0.0  0.0   0.0   2.0   0.0  1.0   1.0   
8  0.0  10.0   30.0  0.0   0.0  0.0  0.0  0.0   0.0   0.0   0.0  0.0   0.0   

   수영장  수영장  궁도장  야구장  축구장  농구장  배구장  배드민턴장 골프연습장  게이트볼 체력단련시설  동물원  식물원  \
4  0.0  0.0  0.0  0.0  0.0  2.0  5.0    0.0     0   0.0      3  0.0  0.0   
5  0.0  0.0  0.0  0.0  0.0  2.0  0.0    2.0     0   0.0      6  0.0  0.0   
6  0.0  0.0  0.0  0.0  1.0  0.0  0.0    1.0     0   2.0      9  0.0  0.0   
7  0.0  0.0  0.0  0.0  0.0  1.0  0.0    2.0     0   1.0      2  0.0  0.0   
8  0.0  0.0  0.0  0.0  0.0  0.0  0.0    0.0     0   0.0      -  0.0  1.0   

   도서관독서실  문화회관  야외음악당 동상  탑기념비  주차장  휴게음식점   매점  자판기  화장실 발효식화장실  간이화장실 공중전화  \
4     0.0   0.0    0.0  1   5.0  1.0    0.0  1.0  0.0  5.0      0    0.0    2   
5     1.0   0.0    1.0  0   0.0  0.0    0.0  0.0  0.0  1.0      0    0.0    -   
6     0.0   0.0    0.0  0   0.0  0.0    0.0  0.0  0.0  2.0      0    0.0    0   
7     0.0   0.0    1.0      0.0  2.0    1.0  2.0  0.0  7.0      0    0.0    4   
8     0.0   0.0    0.0  0   0.0  0.0    0.0  1.0  0.0  3.0      0    0.0    1   

   음수대   샘터  시계탑  
4  7.0  0.0  0.0  
5  2.0  0.0  0.0  
6  2.0  0.0  0.0  
7  7.0  0.0  1.0  
8  1.0  0.0  0.0

# 공원별

In [ ]:
# main_namsan = main[main['공원명'] == '남산공원']
# park_namsan = park[park['공 원 명'] == '남 산']
# facilities_namsan = facilities[facilities['공원명'] == '남   산']

1. 남산공원 - 남산
2. 길동생태공원 - (일자산)
3. 서울대공원 
4. 서울숲 
5. 월드컵공원
6. 광화문시민열린마당 
13. 여의도근린공원? 
17. 보라매근린공원
18. 어린이대공원
19. 천호근린공원 - 천호
20. 용산가족공원
22. 낙산근린공원 - 낙산
23. 매헌시민의숲 - 시민의 숲
86. 선유도근린공원
91. 북서울꿈의숲
92. 서서울호수공원
93. 중랑캠핑숲
107. 세종로공원
110. 푸른수목원
128. 경춘선 숲길
130. 서울로7017 
131. 서울식물원